code/modules/telesci/gps.dm

```diff
@@ -34,6 +34,8 @@ var/list/GPS_list = list()
 /obj/item/device/gps/AltClick(mob/user)
 	if(!in_range(src, user))
 		return
+	if(!user.canUseTopic(user))

```

Comment:
also ref #17839

also I wasn't quite clear, but you don't need to use in_range if you're using canUseTopic, set be_close=TRUE and it will do the range/tk checks for you